index.js

```diff
@@ -105,10 +106,16 @@ function patchOnRead(ctx) {
 
 wrap(net.Socket.prototype, 'connect', function (original) {
   return function () {
+    var args;
+    if (v8plus && Array.isArray(arguments[0])) {
```

Comment:
Here's the related code in Node 8:

https://github.com/nodejs/node/blob/0d8021e5a4cf0a6aa3a700a361f6d42c2894f2ba/lib/net.js#L943-L951

As you can see Node core uses an internal symbol referenced by `normalizedArgsSymbol` (which we don't have access to) as an extra guard in case someone actually by a error passes in an array as the first argument. First I thought we couldn't check for at as the `normalizedArgsSymbol` symbol isn't available to us, but I just realized that we could add a check for if there's a symbol or not - which might be better than nothing:

```js
if (v8plus && Array.isArray(arguments[0]) && Object.getOwnPropertySymbols(arguments[0]).length > 0) {
```

We could also go all the way to to a string comparison, but I feel that's way too dirty:

```js
Object.getOwnPropertySymbols(arguments[0]).toString() === 'Symbol(normalizedArgs)'
```

I could also be fine with just merging the PR as is. But I just wanted to follow this train of thoughts 😅 

@Qard what do you think?

Reply:
I think it might have `Symbol.iterator` on it, so just checking for presence of *any* symbols probably won't be helpful. We can probably just drop that check and just go by if it's an array.

Reply:
so, just keep it the same as this is acceptable?
```
if(v8plus && Array.isArray(arguments[0])) {
```

Reply:
An normal array will not have a Symbol, so this should still guard against that case. So I think I'll merge this PR in as is and then add a comment about the Symbol gotcha next to the if-sentence. Thanks for the patience 😄